File: C/tda.c
#include <stdio.h>
#include <stdlib.h>

int gcd(int a, int b) {
    if (b == 0) {
        return a;
    } else {
        return gcd(b, a % b);
    }
}

int main(void){
    int n = 0; 
    int N1, N2, D1, D2;
    char DIV1, DIV2, OP;
    scanf("%d", &n);
    
    for(int i = 0; i < n; i++){
        scanf("%d %c %d %c %d %c %d", &N1, &DIV1, &D1, &OP, &N2, &DIV2, &D2);
        int numerator, denominator;
        char result[100], simpleResult[100];
        
        if(OP == '+'){
            numerator = N1 * D2 + N2 * D1;
            denominator = D1 * D2;
        } else if(OP == '-'){
            numerator = N1 * D2 - N2 * D1;
            denominator = D1 * D2;
        } else if(OP == '*'){
            numerator = N1 * N2;
            denominator = D1 * D2;
        } else if(OP == '/'){
            numerator = N1 * D2;
            denominator = N2 * D1;
        }
        
        // Simplificar a fração
        int gcdValue = gcd(numerator, denominator);
        numerator /= gcdValue;
        denominator /= gcdValue;
        
        // Formatar a expressão original
        sprintf(result, "%d%c%d %c %d%c%d", N1, DIV1, D1, OP, N2, DIV2, D2);
        
        // Formatar a fração simplificada
        sprintf(simpleResult, "%d/%d", numerator, denominator);
        
        // Exibir a expressão original e a fração simplificada
        printf("%s = %s\n", result, simpleResult);
    }
    
    return 0;
}

File: Java/tda.java
import java.util.Scanner;

public class tda {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n;
        n = sc.nextInt();
        for(int i = 0; i < n; i++){
            int n1, n2, d1, d2;
            String div1, div2;
            char op;
            n1 = sc.nextInt();
            div1 = sc.next();
            d1 = sc.nextInt();
            op = sc.next().charAt(0);
            n2 = sc.nextInt();
            div2 = sc.next();
            d2 = sc.nextInt();
            String result = "";
            String simpleResult = "";
            if(op == '+'){
                result = (n1*d2 + n2*d1) + "/" + (d1*d2);
                int numerator = n1*d2 + n2*d1;
                int denominator = d1*d2;
                int gcd = 1;
                for (int j = 1; j <= numerator && j <= denominator; j++) {
                    if (numerator % j == 0 && denominator % j == 0) {
                        gcd = j;
                    }
                }
                simpleResult = (numerator / gcd) + "/" + (denominator / gcd);
            } else if(op == '-'){
                result = (n1*d2 - n2*d1) + "/" + (d1*d2);
                int numerator = n1*d2 - n2*d1;
                int denominator = d1*d2;
                int gcd = 1;
                for (int j = 1; j <= Math.abs(numerator) && j <= denominator; j++) {
                    if (numerator % j == 0 && denominator % j == 0) {
                        gcd = j;
                    }
                }
                simpleResult = (numerator / gcd) + "/" + (denominator / gcd);
            } else if(op == '*'){
                result = (n1*n2) + "/" + (d1*d2);  
                int numerator = n1*n2;
                int denominator = d1*d2;
                int gcd = 1;
                for (int j = 1; j <= numerator && j <= denominator; j++) {
                    if (numerator % j == 0 && denominator % j == 0) {
                        gcd = j;
                    }
                }
                simpleResult = (numerator / gcd) + "/" + (denominator / gcd);
            } else if(op == '/'){
                result = (n1*d2) + "/" + (n2*d1);  
                int numerator = n1*d2;
                int denominator = n2*d1;
                int gcd = 1;
                for (int j = 1; j <= numerator && j <= denominator; j++) {
                    if (numerator % j == 0 && denominator % j == 0) {
                        gcd = j;
                    }
                }
                simpleResult = (numerator / gcd) + "/" + (denominator / gcd);
            }
            System.out.println(result + " = " + simpleResult);
        }
        sc.close();
    }
}
